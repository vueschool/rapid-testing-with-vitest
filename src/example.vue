<script setup>
const items = [
  'apple',
  'banana',
  'orange'
]

const current = ref(items[0])

function next() {
  const index = items.indexOf(current.value)
  current.value = items[(index + 1) % items.length]
}

function prev() {
  const index = items.indexOf(current.value)
  current.value = items[(index - 1 + items.length) % items.length]
}
</script>

<template>
  <div>
    <button @click="prev">Prev</button>
    <div>{{ current }}</div>
    <button @click="next">Next</button>
  </div>
</template>
