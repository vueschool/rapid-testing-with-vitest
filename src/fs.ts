import { readFileSync, existsSync } from 'fs'

export function loadConfig() {
  if (!existsSync('./config.json'))
    return undefined
  return JSON.parse(readFileSync('./config.json', 'utf-8'))
}
