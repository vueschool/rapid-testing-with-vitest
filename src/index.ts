export function deepMerge(a, b) {
  if (Array.isArray(a) && Array.isArray(b)) {
    return [...a, ...b]
  }
  if (Array.isArray(a) || Array.isArray(b) || typeof a !== typeof b) {
    throw new Error('Can not merge two differnet types')
  }
  const merged = { ...a }
  for (const key of Object.keys(b)) {
    if (typeof a[key] === 'object' || Array.isArray(a[key]))
      merged[key] = deepMerge(a[key], b[key])
    else
      merged[key] = b[key]
  }
  return merged
}
