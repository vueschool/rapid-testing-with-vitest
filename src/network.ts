import { fetch } from 'ofetch'

export async function getPostBody(id:number) {
  const data = await fetch(
    'https://jsonplaceholder.typicode.com/posts/' + id
  ).then(r => r.json())

  return data.body
}
