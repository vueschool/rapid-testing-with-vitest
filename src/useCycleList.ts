import { ref } from 'vue'

export function useCycleList(items) {
  const current = ref(items[0])

  function next() {
    const index = items.indexOf(current.value)
    current.value = items[(index + 1) % items.length]
  }

  function prev() {
    const index = items.indexOf(current.value)
    current.value = items[(index - 1 + items.length) % items.length]
  }

  return {
    current,
    next,
    prev
  }
}
