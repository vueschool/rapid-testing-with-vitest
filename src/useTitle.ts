import { ref, watch } from 'vue'

export function useTitle() {
  const title = ref(document.title)
  
  watch(title, () => {
    document.title = title.value
  }, { flush: 'sync' })

  return title
}
