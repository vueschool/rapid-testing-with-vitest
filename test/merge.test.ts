import { test, expect } from 'vitest'
import { deepMerge } from '../src'

test('shallow merge', () => {
  const merged = deepMerge(
    {
      name: 'Anthony'
    },
    {
      github: 'antfu'
    }
  )

  expect(merged).toEqual({
    name: 'Anthony',
    github: 'antfu'
  })
})

test('shallow merge with overlaps', () => {
  const merged = deepMerge(
    {
      name: 'Anthony',
      github: 'unknown'
    },
    {
      github: 'antfu',
      twitter: 'antfu7'
    }
  )

  expect(merged).toEqual({
    name: 'Anthony',
    github: 'antfu',
    twitter: 'antfu7'
  })
})


test('shallow merge with arrays', () => {
  const merged = deepMerge(
    ['vue', 'react'],
    ['svelte', 'solid']
  )

  expect(merged).toEqual([
    'vue', 'react', 'svelte', 'solid'
  ])
})
