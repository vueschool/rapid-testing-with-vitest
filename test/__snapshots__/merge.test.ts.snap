// Vitest Snapshot v1

exports[`deep merge with overlaps 1`] = `
{
  "accounts": {
    "github": "unknown",
    "twitter": "antfu7",
  },
  "name": "Anthony",
}
`;
